gazelle/python/testdata/per_file_non_empty_init/BUILD.out

```diff
@@ -0,0 +1,16 @@
+load("@rules_python//python:defs.bzl", "py_library")
+
+# gazelle:python_generation_mode file
+
+py_library(
+    name = "__init__",
```

Comment:
I am wondering about the convention here and how the targets would look like where we are using them. Having `//foo/bar:__init__` is not the nicest label at the call site, at the same time all of the Pythonistas will understand what is being meant by the label.

Reply:
yeah. I thought about just `:init`, but then what if you have an `init.py`?

if you have anything in the file, everything underneath is probably considered an implementation detail (or you're intentionally doing some side-effect on import 😨) so there's a good chance you want one of the other generation modes or you're gonna manually create a public target for `__init__.py`

Reply:
I guess the ideal naming for the `__init__.py` target IMHO should be as close to the import as possible (e.g. `//foo/bar`) but that would have a different semantic here. an alternative would be to have the following naming rules:
```
src.py -> name = "src_py"
__init__.py -> name = <package_name>
```

That way we could have `//foo/bar` for an import `foo.bar` and `//foo/bar:baz_py` for an import `foo.bar.baz`. The `_py` suffix is somewhat annoying, but was wondering about what others thought about it.

Reply:
most python projects I interact with have empty `__init__.py`s, so gazelle wouldn't generate targets for them and we'd just have a bunch of pointless `_py`s

I'd prefer to leave it as is in this PR and add an option if anyone cares about their `__init__.py`s